### PR TITLE
Added rsplit() for String class

### PR DESCRIPTION
**Pull request by Krakean**
Script used for testing:
...

	var some_string = "Milk, Chicken, Bread, Butter"

	#var t0 = some_string.rsplit(", ", false, 0)  # ['Milk', 'Chicken', 'Bread', 'Butter']
	#var t1 = some_string.rsplit(", ", false, 2)  # ['Milk, Chicken', 'Bread', 'Butter']
	
	var t0 = some_string.rsplit(" ", false, 0)  # ['Milk,', 'Chicken,', 'Bread,', 'Butter']
	var t1 = some_string.rsplit(" ", false, 2)  # ['Milk, Chicken,', 'Bread,', 'Butter']

	print("source: " + some_string)
	print("====")

	for word in t0:
		print("t0: " + word)	
	
	print("====")

	for word in t1:
		print("t1: " + word)

...

### CODE REVIEW

**Comment by Noshyaar** [comment]
>`",amet sit dolor ipsum lorem"`

Why are words in a reversed order? I don't think that's how it is in Python.

**Comment by BeayemX** [comment]
Why does the comma switch from being after the word `amet` to being in front of the word?

Why don't you use `["one", "two", "three", ...]` for showcases like this. I think this would make it easier to see what happens exactly and to spot mistakes.

**Comment by Krakean** [comment]
@Noshyaar correction done. First message is edited with updated script test example. Should be fine now, I suppose

**Comment by Noshyaar** [comment]
>`var t0 = some_string.rsplit(", ", false, 0)`
`['Milk', ' Chicken', ' Bread', ' Butter']`

A space in front of the last three words is not actually there? It could be unexpected for users.

**Comment by Krakean** [comment]
@Noshyaar agree. Corrected. Check, now it is ok?

File: doc/classes/String.xml [hunk 1]
```diff
@@ -666,6 +666,20 @@
...
+			<return type="PoolStringArray">
+			</return>
+			<argument index="0" name="divisor" type="String">
+			</argument>
+			<argument index="1" name="allow_empty" type="bool" default="True">
+			</argument>
+			<argument index="2" name="maxsplit" type="int" default="0">
+			</argument>
+			<description>
+				Splits the string by a [code]divisor[/code] string and returns an array of the substrings, starting from right. Example "One,Two,Three" will return ["Three","Two","One"] if split by ",".
```
Comment (Noshyaar) [inline doc/classes/String.xml]: >`["Three","Two","One"]`

It is in a reversed order.

File: doc/classes/String.xml [hunk 2]
```diff
@@ -666,6 +666,20 @@
...
+                If [code]maxsplit[/code] is specified, then it is number of splits to do, default is 0 which splits all the items.
```
Comment (Noshyaar) [inline doc/classes/String.xml]: This line should be indented with tabs. Somehow this affects all your PRs involving docs at the 2nd line description.

**Comment by Krakean** [comment]
@Noshyaar corrected.

**Comment by Krakean** [comment]
@akien-mga 
Replaced with #14715 to solve merge conflicts.

**Pull request closed without merging**